src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -89,6 +90,15 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  do_unmap();
```

Comment:
@trociny Sorry that I don't catch your idea very much .... could you talk more about the potential data race? Thanks!